src/test/java/org/junitpioneer/testkit/assertion/single/TestCaseFailureAssert.java

```diff
@@ -30,4 +33,16 @@
 	 */
 	AbstractThrowableAssert<?, ? extends Throwable> withException();
 
+	/**
+	 * Asserts that the test/container threw an exception that fulfills the supplied predicate.
+	 * @param predicate the condition the thrown exception must fulfill
+	 */
+	void withExceptionFulfilling(Predicate<Throwable> predicate);
+
+	/**
+	 * Applies the supplied consumer to the exception thrown by the test/container.
+	 * @param testFunction a consumer, for writing more flexible tests
+	 */
+	void andThenCheckException(Consumer<Throwable> testFunction);
```

Comment:
I really have difficulty to imagine how to use this. I think it will quite be complicated and not easy to use - what's a thing I expect of an assertion lib (or method).

Reply:
This is for cases that can not be covered by the other asserting methods.